pkg/drain/filters.go

```diff
@@ -161,63 +170,64 @@ func (d *Helper) daemonSetFilter(pod corev1.Pod) podDeleteStatus {
 	// Such pods will be deleted if --force is used.
 	controllerRef := metav1.GetControllerOf(&pod)
 	if controllerRef == nil || controllerRef.Kind != appsv1.SchemeGroupVersion.WithKind("DaemonSet").Kind {
-		return makePodDeleteStatusOkay()
+		return makePodDeleteStatusOkay(podDeletionPriorityHighest)
 	}
 	// Any finished pod can be removed.
 	if pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed {
-		return makePodDeleteStatusOkay()
+		return makePodDeleteStatusOkay(podDeletionPriorityHighest)
 	}
 
 	if _, err := d.Client.AppsV1().DaemonSets(pod.Namespace).Get(controllerRef.Name, metav1.GetOptions{}); err != nil {
 		// remove orphaned pods with a warning if --force is used
 		if apierrors.IsNotFound(err) && d.Force {
-			return makePodDeleteStatusWithWarning(true, err.Error())
+			return makePodDeleteStatusWithWarning(true, err.Error(), podDeletionPriorityHighest)
 		}
 
 		return makePodDeleteStatusWithError(err.Error())
 	}
 
 	if !d.IgnoreAllDaemonSets {
-		return makePodDeleteStatusWithError(daemonSetFatal)
+		//updating this to set the status as OK if we intend to delete daemon sets
+		return makePodDeleteStatusOkay(podDeletionPriorityLowest)
 	}
 
-	return makePodDeleteStatusWithWarning(false, daemonSetWarning)
+	return makePodDeleteStatusWithWarning(false, daemonSetWarning, podDeletionPriorityLowest)
```

Comment:
Why should this and the previous be `Lowest`? Better would be `podDeletionPriorityDaemonset`.